cmd/config.go

```diff
@@ -249,21 +250,130 @@ func (d *ConfigDuration) UnmarshalYAML(unmarshal func(interface{}) error) error
 	return nil
 }
 
-// GRPCClientConfig contains the information needed to talk to the gRPC service
+// GRPCClientConfig contains the information setup a gRPC client. The following
+// combination of fields are allowed:
+//
+// ServerIPAddresses, [Timeout]
+// ServerAddress, [Timeout], [DNSAuthority]
+// SRVLookup, [Timeout], [DNSAuthority]
 type GRPCClientConfig struct {
-	// ServerAddress is a single host:port combination that the gRPC client
-	// will, if necessary, resolve via DNS and then connect to. This field
-	// cannot be used in combination with `ServerIPAddresses` field.
+	// DNSAuthority is a single <hostname|IPv4|[IPv6]>:<port> or `:<port>` of
+	// the DNS server to be used for resolution of gRPC backends. If the address
+	// contains a hostname the gRPC client will resolve it via the system DNS.
+	// If the address contains a port, the client will use it directly,
+	// otherwise port 53 is used.
+	DNSAuthority string
+
+	// SRVLookup contains the service and domain name the gRPC client will use
+	// to construct a SRV DNS query to lookup backends. For example: if the
+	// resource record is 'foo.service.consul', then the 'Service' is 'foo' and
+	// the 'Domain' is 'service.consul'. The expected dNSName to be
+	// authenticated in the client certificate would be 'foo.service.consul'.
```

Comment:
This is the client config, so it will be authenticating the server's certificate.

```suggestion
	// the 'Domain' is 'service.consul'. The expected dNSName to be
	// authenticated in the server certificate would be 'foo.service.consul'.
```